src/php7/solr_macros.h

```diff
@@ -170,6 +170,49 @@ static inline solr_ustream_t *solr_get_ustream_object(zend_object *obj)
 
 #define Z_USTREAM_P(zv) solr_get_ustream_object(Z_OBJ_P(zv));
 
+/**
+ * Handles strict/non-strict integer parsing
+ *
+ * usage:
+ * 		SOLR_PARSE_PARAM_INT_DEF(maxSegmentsZval, maxSegments, maxSegmentsLen, "1");
+ * 		if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &maxSegmentsZval) == FAILURE) {
+ *			RETURN_THROWS();
+ *		}
+ *		SOLR_PARSE_PARAM_INT(maxSegmentsZval, maxSegments, maxSegmentsLen, "maxSegments");
+ */
+#define SOLR_PARSE_PARAM_INT_DEF(zval_ptr, solrparam, solrparam_len, default_value) \
```

Comment:
I don't like the name of this macro, "DEF" sounds too much like "default". But I don't like `SOLR_PARSE_PARAM_INT_DEFINITION` either :confused: 

`SOLR_DEFINE_PARSE_PARAM_INT` ? (not much better)